core/bind/core_bind.h

```diff
@@ -143,6 +143,14 @@ class _OS : public Object {
 		MONTH_DECEMBER
 	};
 
+	enum HandleType {
+		APPLICATION_HANDLE, // HINSTANCE, NSApplication*, UIApplication*, JNIEnv* ...
+		DISPLAY_HANDLE, // X11::Display* ...
+		WINDOW_HANDLE, // HWND, X11::Window*, NSWindow*, UIWindow*, Android activity ...
+		WINDOW_VIEW, // HDC, NSView*, UIView*, Android surface ...
+		OPENGL_CONTEXT // HGLRC, X11::GLXContext, NSOpenGLContext*, EGLContext* ...
+	};
+
```

Comment:
Is it necessary to define this here since `core_bind.h` includes `os.h` which also contains a similar `HandleType` declaration?

Reply:
I think so, I'm not 100% sure but the same is done with a number of other enums.. The problem is that core_bind is a wrapper that exposes OS to GDScript/GDNative and we can only expose constants if they are part of the class being exposed. 

I could be wrong though, I should experiment with it a little more.

Reply:
Actually @akien-mga , can you confirm this? Do enum defs have to be in the same class for the bind macros to work?

Reply:
I think so, it's probably not possible to bind enums which are not defined in the current class as part of that class' API.
Might have improved with @reduz's recent refactoring of method binding, not sure.

Reply:
Well that would be Godot 4 anyway, so we could revisit it when we port this to 4